### PR TITLE
contracts-bedrock: L1Block interop

### PR DESCRIPTION
**Pull request by tynes**
**Description**

Ports the `L1Block` contract from the `feat/interop` branch to
develop using a pattern where we can extend the contracts.
This sort of pattern may not always work but is an experiment
in reducing feature branches.

### CODE REVIEW

**Comment by tynes** [comment]
Looks like this needs some linting

File: packages/contracts-bedrock/test/L2/L1BlockInterop.t.sol [hunk 1]
```diff
@@ -0,0 +1,259 @@
...
+    ///      the length of _dependencySet. (happy path)
+    function testFuzz_setL1BlockValuesInterop_dependencySetSizeMatch_succeeds(uint256[] calldata dependencySet)
+        external
+    {
+        vm.assume(dependencySet.length <= type(uint8).max);
+
+        bytes memory functionCallDataPacked = Encoding.encodeSetL1BlockValuesInterop({
+            _baseFeeScalar: type(uint32).max,
+            _blobBaseFeeScalar: type(uint32).max,
+            _sequenceNumber: type(uint64).max,
```
Comment (tynes) [inline packages/contracts-bedrock/test/L2/L1BlockInterop.t.sol]: This test doesn't give additional coverage on top of `testFuzz_setL1BlockValuesInterop_succeeds`

**Comment by tynes** [comment]
This looks good to me given a lint. I cannot approve since I opened the PR

**Review by 0xfuturistic — APPROVED** [review]

**Comment by tynes** [comment]
@ethereum-optimism/contract-reviewers would love a review here

**Review by 0xfuturistic — APPROVED** [review]

File: packages/contracts-bedrock/test/L2/L1BlockInterop.t.sol [hunk 2]
```diff
@@ -0,0 +1,234 @@
...
+
+    /// @dev Tests that `setL1BlockValuesInterop` reverts if _dependencySetSize is not the same as
+    ///      the length of _dependencySet. (bad path)
+    function testFuzz_setL1BlockValuesInterop_dependencySetSizeMatch_reverts(
+        uint8 _notDependencySetSize,
+        uint256[] calldata _dependencySet
+    )
+        external
+    {
+        vm.assume(_dependencySet.length <= type(uint8).max);
```
Comment (pegahcarter) [inline packages/contracts-bedrock/test/L2/L1BlockInterop.t.sol:129]: How was this max determined?

Comment (tynes) [reply packages/contracts-bedrock/test/L2/L1BlockInterop.t.sol:129]: Simplicity as it's 1 byte

**Pull request merged**